### ISSUE TITLE
kpasswd not working with LDAP: db backend

### ISSUE THREAD

**Issue by tipt0e** [heimdal/heimdal#334]
It is impossible to change a principal's password using the LDAP: db backend with heimdal 7.4.0.  The issue is similar to the earlier one I opened (which heretofore I have received _no_ response), except this one is more severe in that password changes are affected.
Example:
`rochelle@radium:~$ date
Sun Sep 17 07:48:10 PDT 2017 <--- showing time of password change attempt
rochelle@radium:~$ kpasswd
rochelle@BYTEPIMPS.NET's Password:   <--- correct password
New password: 
Verify password - New password: 
Auth error : Authentication failed <--- ???
rochelle@radium:~$
***ERROR MESSAGES:***
root@radium:~# tail -2 /var/log/krb.log 
2017-09-17T07:48:23 krb5_rd_req: start_seq_get is not supported in the HDBGET keytab type
2017-09-17T07:48:23 krb5_rd_req: start_seq_get is not supported in the HDBGET keytab type`

Can someone please take a look at this?  I had a change password facility in place for my email users and now it is hopelessly broken.

**Comment by tipt0e** [comment]
I don't know why words were crossed out above, but please read through the cross-out.

**Comment by jaltman** [comment]
> Can someone please take a look at this? I had a change password facility in place for my email users and now it is hopelessly broken.

Please be aware that Heimdal development is purely on a volunteer basis by members of the community.  There is no paid staff and response to issues is entirely conditional upon the availability of volunteers to reproduce the problem, analyze the issue, and develop a fix.  

Failure to respond is simply an indication that no volunteer has time to reproduce the environment in which the problem has been reported.

**Comment by tipt0e** [comment]
Understood, but I will point out that this is about 99% aligned with https://github.com/heimdal/heimdal/issues/133
and somewhat aligned with:
https://github.com/heimdal/heimdal/issues/41

Just a thought, thanks.

**Comment by nicowilliams** [comment]
The messages you included are very helpful, I think.

Basically, the HDB system can act like a keytab, which is nice so you don't have to setup keytabs for KDC services like kpasswdd.  If a service cannot obtain a key from a keytab with which to decrypt a Ticket presented by the client, the service will attempt to use all keys (of the right enctype) from the keytab, just in case (this is very useful).  Obviously that's not something one would want for a KDC service that is using the HDB as a keytab!  So the HDB does not support iteration like the FILE keytab type does.

What this means here, most likely, is that you did not create the `kadmin/changepw` principal.  Can you check if you have this principal?

**Comment by tipt0e** [comment]
Yes, the principal was created when I did the initial 'init BYTEPIMPS.NET'.

**Comment by tipt0e** [comment]
Here's the listing:

root@radium:~# kadmin -l
kadmin> list kadmin/changepw
kadmin/changepw
kadmin>

**Comment by tipt0e** [comment]
Some more info:

root@radium:~# kadmin -K /root/chgpw.keytab -p chgpw passwd rochelle
rochelle@BYTEPIMPS.NET's Password: 
Verify password - rochelle@BYTEPIMPS.NET's Password: 
kadmin: cpw rochelle: Generic error (see e-text)
root@radium:~# tail -1 /var/log/krb.log 
2017-09-17T20:47:08 krb5_recvauth: start_seq_get is not supported in the HDBGET keytab type
root@radium:~# cat /var/heimdal/kadmind.acl 
mikal@BYTEPIMPS.NET     all
chgpw@BYTEPIMPS.NET     cpw

**Comment by tipt0e** [comment]
I was able to get around this issue for now by setting an explicit keytab for kpasswdd:

/usr/libexec/kpasswdd -k /path/to/chgpw.keytab --detach

Closing this because I have it working, but the 'start_seq_get is not supported in the HDBGET keytab type' error when using the ldap backend still exists, when it didn't before 7.x.

**Comment by nicowilliams** [comment]
Well, I'm glad that worked for you, but that should not have been necessary.

**Comment by tipt0e** [comment]
I am sorry (jaltman) but has there been any movement on this? I only asked because it was bugged.  I am loath to provide a patch because of some of the behavior on this issues list.

**Comment by tipt0e** [comment]
and I am using 7.5.0

**Comment by nicowilliams** [comment]
I'm completely mystified.  I'll have to write a test.
